### ISSUE TITLE
While multi-column markdown is active => Lag

### ISSUE THREAD

**Issue by GottaBeTheJuice** [ckRobinson/multi-column-markdown#63]
Lag and slowing on typing even on notes without any multi-column use.

**Suggestion**
Haven't looked at the code, but it seems like it would be great if there was a way to tell the main workload of the plugin to activate only when a certain YAML activator or function on the note is invoked. I do not mind one bit it being a little laggy on my dashboard , but on the notes it is frustrating.

Anyone else experience this?

**Comment by ckRobinson** [comment]
After a bit of investigating I found this bug. It seems, at least on my end, that depending on system specs it may not be noticeable in most cases but once file sizes get large enough, and certain kinds of markdown content is added, Obsidian begins to hang on typing or clicking within the file.

I have implemented a fix that has resolved the issue in my test cases and it will be a part of the next update which I hope to release soon.

**Comment by GottaBeTheJuice** [comment]
Great! I'd be curious to know what was causing it and what the remedy was in this type of situation! My system is beefy so probably just an inherit problem when, as you said, file sizes get large enough or certain markdown is added like images maybe.  I know you said you have solved it in the coming update, but for reference purposes: 

My vault is "only" about 360 files, with about 300 links and 18,000 words. There are also 107 attachments, most of which are images, maybe 2 videos I screen recorded and attached. 

Great work on cranking out the bug fix, look forward to re-activating the plugin when its released _AND seeing what the fix was and how that logic patched the crack in performance!_

**Comment by ckRobinson** [comment]
It's actually a fairly simple fix, just slipped by my initial live preview implementation.

Basically when live preview renders a file it passes a list of "changes" to the plugin on every mouse click or keystroke. Normally there are only a few changes in the list per refresh so everything runs smoothly. In my testing however I found that some files and content, specifically large codeblocks, were sending thousands of changes at once. And then for each change the plugin was checking if it needed to render a column region which dragged Obsidian to a crawl. The fix was just to add an extra flag and make sure that live preview ignores any unnecessary calculations.

Anyways the fix has now been released in 0.7.7 and should hopefully fix the problem for you. If the problem persists on your machine there may be an additional bug somewhere in there. I have extra optimizations that I could implement but will take time to get things right so hoping this initial fix solves the problem for you. If the problem does persist I'll re-open the issue.
